ompi/communicator/comm_cid.c

```diff
@@ -963,6 +1047,73 @@ int ompi_comm_activate (ompi_communicator_t **newcomm, ompi_communicator_t *comm
     return rc;
 }
 
+int ompi_comm_get_remote_cid (ompi_communicator_t *comm, int dest, uint32_t *remote_cid)
+{
+    ompi_proc_t *ompi_proc;
+    pmix_proc_t  pmix_proc;
+    pmix_info_t tinfo[2];
+    pmix_value_t *val = NULL;
+    ompi_comm_extended_cid_t excid;
+    int rc = OMPI_SUCCESS;
+    size_t remote_cid64;
+
+    assert(NULL != remote_cid);
+
+    if (OPAL_LIKELY(OMPI_COMM_IS_GLOBAL_INDEX(comm))) {
+
+        *remote_cid = comm->c_index;
+
+    } else if (0 != comm->c_index_vec[dest]) {
+
+        *remote_cid = comm->c_index_vec[dest];
```

Comment:
IMO we should extractr the "likely" code path of this function (the first 2 cases where comm is global or the cid is initialized) to an inline function, since it's called on the fast path of every send operation.
Also, maybe returning the cid value / negative error code as ssize_t/int64_t (instead of by-value *remote_cid) may help the compiler to optimize the code.